XenAdmin/MainWindow.cs

```diff
@@ -729,28 +729,18 @@ internal void ProcessCommand(params string[] args)
                             Password = args.Length > 3 ? args[3] : ""
                         };
 
-                        if (ConnectionsManager.XenConnectionsContains(connection))
-                            break;
-
-                        lock (ConnectionsManager.ConnectionsLock)
-                            ConnectionsManager.XenConnections.Add(connection);
-
-                        XenConnectionUI.BeginConnect(connection, true, null, false);
+                        if (File.Exists(args[1]))
+                            XenConnectionUI.ConnectToXapiDatabase(connection, this);
+                        else
+                            XenConnectionUI.BeginConnect(connection, true, this, false);
                         break;
                     default:
                         log.Warn("CLI: Wrong syntax or unknown command line options.");
                         break;
                 }
             }
 
-            if (_launched)

```

Comment:
I think `_launcehd` is not needed because this method is called only when the MainWindow has been created.